design/pipeline-triggers.md

```diff
@@ -0,0 +1,369 @@
+  repositories:
+  - repository: myPHPApp      
+    type: GitHub
+    connection: myGitHubConnection
+    source: ashokirla/phpApp
+    pr: 
+    - master
+    - releases/*
+```
+
+You can specify the branches and file paths to include and exclude.
+```yaml
+resources:         
+  repositories:
+  - repository: myPHPApp      
+    type: GitHub
+    connection: myGitHubConnection
+    source: ashokirla/phpApp
+    pr:
+      branches:
+        include:
+        - releases/*
+        exclude:
+        - releases/experimental/*
+      paths:
+        include:
+        - web/*
+        exclude:
+        - web/README.md
+```
+
+
+
+### Containers
+Whenever a new image got published to the container registry, your pipeline run will be triggered. See [container resource](pipeline-resources.md#resources-containers) for more details.
+
+
+#### Scenarios	
+- I would like to trigger my pipeline whenever a new image got published.
+- I would like to trigger my pipeline whenever a new image got published to ‘East-US’ location (ACR specific filter).
+
+
+#### Schema
+```yaml
+resources:          
+  containers:
+  - container: string       
+    type: enum  
+    connection: string 
+    image: string # container image name, Tag/Digest is optional; defaults to latest image
+    trigger: # Optional; Triggers are enabled by default
+      tags:
+        include: [ string ]  # image tags to consider the trigger events, optional; defaults to any new tag
+        exclude: [ string ]  # image tags on discard the trigger events, option; defaults to none
+      location: [ string ]  # geo-location the image published to; ACR specific setting.
+```
+
+#### Examples
+
+Triggers are enabled by default on the `container` resource. When a new image gets published to your image registry, pipeline run starts automatically. You can disable the triggers on your `container`.
+
+```yaml
+resources:         
+  containers:
+  - container: smartHotel 
+    type: Docker
+    connection: myDockerRegistry
+    image: smartHotelApp 
+    trigger: none
+```
+
+
+You can specify the image tag format to get the trigger by simple syntax.
+```yaml
+resources:         
+  containers:
+  - container: smartHotel 
+    type: Docker
+    connection: myDockerRegistry
+    image: smartHotelApp 
+    trigger:
+    - version-*
+```
+
+
+You can specify the image tags to include and exclude.
+```yaml
+resources:         
+  containers:
+  - container: smartHotel 
+    type: Docker
+    connection: myDockerRegistry
+    image: smartHotelApp 
+    trigger:
+      tags:
+        include:
+        - version-*
+        exclude:
+        - version-2017*
+```
+
+
+If you have an ACR `container` resource, you can specify the geo location to get the triggers.
+```yaml
+repositories:
+  containers:
+  - container: MyACR    
+    connection: RMPM
+    registry: contosodemo
+    image: Microsoft/alphaworz
+    trigger: 
+      tags:
+        include: 
+        - production*
+      location: 
```

Comment:
As I understand it the location data is only part of the payload when we are using ACR and not when coming from DockerHub or Quay.  How are we thinking about addressing that issue?

Reply:
The trigger section maps to the type. For Docker Registry, we don't have location. This property is available only if the type = ACR.